### PR TITLE
fixing non-root compliance and remove git if not required at runtime.…

### PR DESCRIPTION
**Pull request by JakeCodeStuff**
… apt autoremove will remove lingering dependencies no longer attached

## Changes

*Please describe.*  
root user running process is a compliance issue for docker image vulnerability scanning, this alleviates that by creating a posthog user, and making runtime files owned/run by that user. Debian base image uses a disabled root account, so by creating this posthog user, if an attacker were to enter this image process (somehow) they would not be able to elevate privilege to root, as it is disabled. However a system admin can easily still exec into the container as root if needed. Also removed git assuming it is not used at runtime, and would reduce attack surface. apt autoremove removes the lingering dependencies of curl, git, build-essential, etc after those are purged.
*If this affects the frontend, include screenshots.*  

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Review by fuziontech — APPROVED** [review]
lgtm - nice cleanup here!

File: production.Dockerfile [hunk 1]
```diff
@@ -27,5 +27,14 @@ RUN pip install -r requirements.txt --no-cache-dir --compile \
 
 RUN SECRET_KEY='unsafe secret key for collectstatic only' DATABASE_URL='postgres:///' REDIS_URL='redis:///' python manage.py collectstatic --noinput
 
+RUN apt-get purge -y git && apt autoremove -y
```
Comment (Twixes) [inline production.Dockerfile:30]: I suppose `curl` and `build-essentials` are not needed after build either?

Comment (paolodamico) [reply production.Dockerfile:30]: We already purged them, but I did move this to the same line for clarity

**Comment by paolodamico** [comment]
Thanks for the PR @JakeEmo, very helpful! Our apologies for the delayed review, we had quite a bit of backlog. I've gone ahead and created a new PR from this with some minor cleanup and to ensure all tests can run properly. Please email yakko ( a t ) posthog (.) com to get some cool merch, and thanks for the contribution!

**Pull request closed without merging**